### PR TITLE
tools/trap: Refactor TRAP tool

### PR DESCRIPTION
**Pull request by junmin-kim**
Replace duplicated code with loop

### CODE REVIEW

**Comment by junmin-kim** [comment]
Test needed.

File: tools/trap/ramdumpParser.py [hunk 1]
```diff
@@ -820,6 +820,34 @@ def main():
...
+		("elf", elf)
+	] + access_check_list
+
+	for name, path in exist_check_list:
+		if path and not os.path.exists(path):
+			print(f"{path} does not exist. Please provide the proper path for {name}...")
+			sys.exit(1)
+
+	for name, path in access_check_list:
+		if not os.path.exists(readelf_path):
```
Comment (jeongchanKim) [inline tools/trap/ramdumpParser.py:845]: There is a missing about checking permission.
Please check this.

Comment (junmin-kim) [reply tools/trap/ramdumpParser.py]: Fixed it.

**Review by jeongchanKim — APPROVED** [review]
Looks nice to me :) 💯

**Comment by sunghan-chang** [comment]
@thapav Could you review this?

**Review by jeongarmy — APPROVED** [review]
It improves code readability. Looks good.

File: tools/trap/ramdumpParser.py [hunk 2]
```diff
@@ -820,6 +820,34 @@ def main():
 				ext = (line.split("=")[1].strip())
 	elf = (BIN_PATH + 'tinyara' + ext)
 
+	if not log_file and not dump_file:
+		print('Usage error: Must specify one of the -t or -e options. Plz find below for proper usage')
+		usage()
+
+	access_check_list = [
+		("gdb", gdb_path)
```
Comment (thapav) [inline tools/trap/ramdumpParser.py:828]: Missing comma (,) at the end

File: tools/trap/ramdumpParser.py [hunk 3]
```diff
@@ -820,6 +820,34 @@ def main():
...
+
+	exist_check_list = [
+		("dump_file", dump_file),
+		("log_file", log_file),
+		("elf", elf)
+	] + access_check_list
+
+	for name, path in exist_check_list:
+		if path and not os.path.exists(path):
+			print(f"{path} does not exist. Please provide the proper path for {name}...")
```
Comment (thapav) [inline tools/trap/ramdumpParser.py:841]: If we use F-strings, we need to mandate Python 3.6 or above in prerequisites

File: tools/trap/ramdumpParser.py (hunk 2 above)
Comment (junmin-kim) [reply tools/trap/ramdumpParser.py]: Thank you for the review. I fixed it

File: tools/trap/ramdumpParser.py (hunk 3 above)
Comment (junmin-kim) [reply tools/trap/ramdumpParser.py:841]: Python 3.6 reached end-of-life.
And in Ubuntu 20.04 LTS, Python 3.8 is included.
How about mandate Python 3.8 or above in prerequisites?

Comment (thapav) [reply tools/trap/ramdumpParser.py:841]: @sunghan-chang  Shall we go ahead with Python3.8 mandate ? OR
Shall we keep regular print statements across the tool ?

Comment (jeongchanKim) [reply tools/trap/ramdumpParser.py:841]: In TizenRT docker, Python 3.7 is installed. So I think it is better to match with it.
How do you think @thapav  and @junmin-kim ?

Comment (junmin-kim) [reply tools/trap/ramdumpParser.py:841]: Agreed.
FYI) Python 3.7 will reach end-of-life in 2023.

Comment (thapav) [reply tools/trap/ramdumpParser.py:841]: > In TizenRT docker, Python 3.7 is installed. So I think it is better to match with it. How do you think @thapav and @junmin-kim ?

Yes I think that would be better.
@junmin-kim Can you verify F-strings are working on 3.7

Comment (junmin-kim) [reply tools/trap/ramdumpParser.py:841]: I checked that f-string works at 3.7

**Review by thapav — APPROVED** [review]
LGTM

**Review by sunghan-chang — APPROVED** [review]

**Pull request merged**